### PR TITLE
Bazel QD, Refactoring, Stepper Motor lib

### PR DESCRIPTION
**Pull request by bwz5**
This PR refactors the directory hierarchy for the fill station by including sensors and actuators into the fill directory. 

I am unsure if the Bazel refactoring is correct, and I know that this will likely not pass tests as the qd.cc file does not have the correct dependencies linked (Zach help please). 

This PR also establishes WiringPi and SMDriver in the lib directory.

### CODE REVIEW

**Comment by bwz5** [comment]
* **#83** <a href="https://app.graphite.dev/github/pr/cornellrocketryteam/Ground-Software/83?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bwz5 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: fill/actuators/actuator.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#ifndef ACTUATOR_H
+#define ACTUATOR_H
+
+class Actuator {
+public: 
+    /* Actuates the actuator. 
+    Returns true if activation is successful, and false otherwise. 
+    Sets isActuating to true if activation is successful.*/  
+    virtual bool Actuate() = 0;
```
Comment (ZachGarcia42) [inline fill/actuators/actuator.h:9]: I feel like this should be set to false for readability

Comment (bwz5) [reply fill/actuators/actuator.h:9]: I am not sure if that is allowed in my version of the C++ standard. (it wants it to just be =0). I don't know why this is the case, but I am getting the red squiggles in my IDE

Comment (maxslarsson) [reply fill/actuators/actuator.h:9]: I don't think the 0 is the return value if that's what you're thinking. I think it is a special case that when a virtual function = 0, then it has to be implemented by subclasses. I see the = 0 convention a lot in C++

Comment (ZachGarcia42) [reply fill/actuators/actuator.h:9]: Yep, just realized that

File: fill/actuators/actuator.h [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#ifndef ACTUATOR_H
+#define ACTUATOR_H
+
+class Actuator {
+public: 
+    /* Actuates the actuator.
```
Comment (ZachGarcia42) [inline fill/actuators/actuator.h:6]: Can we refactor this file? The current implementation is very difficult to understand. Something like this maybe where active is open for valves or ignited for igniters?

// Starts actuation
virtual bool Actuate() = 0;

// Returns Active
virtual bool isActive() = 0

protected:
bool active;

File: fill/lib/SMDriver/steppermotor.cpp [hunk 3]
```diff
@@ -0,0 +1,181 @@
+#include "steppermotor.h"
+#include <wiringPi.h>
+
+StepperMotor::StepperMotor(int Coil1, int Coil3, bool IsHalfStep)
```
Comment (ZachGarcia42) [inline fill/lib/SMDriver/steppermotor.cpp:4]: Update with additional coils as discussed

File: fill/lib/WiringPi
Comment (ZachGarcia42) [inline fill/lib/WiringPi:1]: This says it's empty. Is that intentional?

**Review by ZachGarcia42 — COMMENTED** [review]
Are the libraries currently git submodules (or bazel repositories rather) and if not, would that be difficult to do?

**Review by maxslarsson — APPROVED** [review]
Ben said every comment was addressed in-person

**Comment by bwz5** [comment]
### Merge activity

* **Nov 4, 6:03 PM EST**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/cornellrocketryteam/Ground-Software/83).

**Pull request merged**